### PR TITLE
Update `spack setup` and `spack graph` to be consistent with recent refactoring

### LINKED ISSUE #2316: `spack setup` does not run
Should be easy to fix, does anyone know what to change here?
```
$ spack setup pism@dev
Traceback (most recent call last):
  File "/home2/rpfische/spack4/bin/spack", line 203, in <module>
    main()
  File "/home2/rpfische/spack4/bin/spack", line 180, in main
    return_val = command(parser, args)
  File "/home2/rpfische/spack4/lib/spack/spack/cmd/setup.py", line 133, in setup
    with spack.installed_db.write_transaction():
AttributeError: 'module' object has no attribute 'installed_db'
```

### PR DESCRIPTION
**Pull request by citibeth**
Fixes #2316.

### CODE REVIEW

**Comment by citibeth** [comment]
@hegner ...

File: lib/spack/spack/directory_layout.py [hunk 1]
```diff
@@ -260,9 +260,10 @@ def check_installed(self, spec):
             return None
 
         if not os.path.isfile(spec_file_path):
-            raise InconsistentInstallDirectoryError(
-                'Install prefix exists but contains no spec.yaml:',
-                "  " + path)
+            return None
+#            raise InconsistentInstallDirectoryError(
+#                'Install prefix exists but contains no spec.yaml:',
+#                "  " + path)
```
Comment (adamjstewart) [inline lib/spack/spack/directory_layout.py]: Is this intended to be part of this PR? This might be a more controversial change.

Comment (tgamblin) [inline lib/spack/spack/directory_layout.py]: It's also fixed as of #2172

Comment (citibeth) [inline lib/spack/spack/directory_layout.py]: #2172 was merged 14 days ago, I'm working with develop from 1 day ago.  So... #2172 has not fixed the problem this change to directory_layout.py is addressing.

This change addresses #2319.  I put it in the same PR that addresses #2316 because both bugs must be addressed in order for `spack setup` to work.  I agree, there might be better ways to fix #2319.  But I don't know of what that way would be.  Absent a better solution ,we should merge this PR, at least temporarily, if we want `spack setup` working today.  I don't want someone at the tutorial trying `spack setup` and having it crash on them very first thing.

If we can find a better way to fix #2319 in the future, I'd be happy to revert this PR and do the better way instead.  Let's not let perfect become the enemy of good enough.

Comment (adamjstewart) [inline lib/spack/spack/directory_layout.py]: Why exactly is this change necessary? Is it only necessary when running `spack setup`? Could we perhaps write a `spec.yaml` for packages installed by `spack setup`?

I think this check needs to stay here for most package installations (via `spack install`).

Comment (citibeth) [inline lib/spack/spack/directory_layout.py]: > Why exactly is this change necessary?

It fixes #2319 

> Is it only necessary when running spack setup?

Yes

> Could we perhaps write a spec.yaml for packages installed by spack setup?

I'm sure there's a better way to fix #2319.  Please see full notes above this comment on the GitHub discussion page (not just in email).

**Comment by pramodk** [comment]
`spack setup` is my favourite feature and use it regularly.
It would be great to have this issue fixed!

**Comment by adamjstewart** [comment]
Yes, that's what's holding it up. Is there any way we could either:

1. Write a `spec.yaml` for packages installed with `spack setup`

or

2. Don't check for `spec.yaml` if the package was installed with `spack setup`

Is there any way of determining whether or not a package was installed with `spack setup`? I assume the error message is being raised when you try to depend on a package installed with `spack setup`, not when you try to run `spack setup` itself?

**Comment by citibeth** [comment]
See #2319 ; I believe this is a timing thing.  Spack eagerly creates an empty install directory, and then later throws an exception.  It should be reproducible when trying `spack setup` on any package, not just when depending on manually built packages.

**Comment by citibeth** [comment]
OK, I reverted that commit.  I couldn't reproduce the error.  Everything else in this PR should be non-controversial and ready to go.  If #2319 rears its ugly head again, we can fix it then.
